packages/plugin/src/rules/unique-fragment-name.ts

```diff
@@ -1,16 +1,64 @@
 import { relative } from 'path';
-import { GraphQLESLintRule } from '../types';
-import { normalizePath, requireSiblingsOperations } from '../utils';
+import { FragmentDefinitionNode, Kind, OperationDefinitionNode } from 'graphql';
+import { GraphQLESLintRule, GraphQLESLintRuleContext } from '../types';
+import { GraphQLESTreeNode } from '../estree-parser';
+import { normalizePath, requireSiblingsOperations, getOnDiskFilepath } from '../utils';
+import { FragmentSource, OperationSource } from '../sibling-operations';
 
+const RULE_NAME = 'unique-fragment-name';
 const UNIQUE_FRAGMENT_NAME = 'UNIQUE_FRAGMENT_NAME';
 
-const rule: GraphQLESLintRule<[], false> = {
+export const checkNode = (
```

Comment:
I removed code duplication in `unique-fragment-name` and `unique-operation-name` rules, moved logic in this `checkNode` function